### PR TITLE
Added a hosts appender command. 

### PR DESCRIPTION
**Pull request by Cyruxx**
ToDo: Test on windows and other operating systems. I could only check it on OSX10

### CODE REVIEW

**Comment by Cyruxx** [comment]
Execute as the following:

`sudo ./homestead 192.168.10.10 homestead.dev`

First argument is the IP address to map and second argument is the hostname.

File: src/HostCommand.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            // ToDo: Someone should test this! I don't have a windows machine.
+            return 'echo "' . $this->ip . ' ' . $this->hostName . '" >> "' . $this->hostFile . '"';
+        } elseif (strpos(strtoupper(PHP_OS), 'DARWIN') === 0
+            || strpos(strtoupper(PHP_OS), 'LINUX') === 0) {
+            return 'echo "' . $this->ip . ' ' . $this->hostName . '" >> "' . $this->hostFile . '"';
+        } else {
+            return false;
+        }
+    }
+}
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: cs

File: src/HostCommand.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    /**
+     * Get the command to execute.
+     *
+     * @return bool|string
+     */
+    protected function getCommand()
+    {
+        if (strpos(strtoupper(PHP_OS), 'WIN') === 0) {
+            // ToDo: Someone should test this! I don't have a windows machine.
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: don't include todo comments please in the code

File: src/HostCommand.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    /**
+     * Find the correct host file depending on the OS.
+     *
+     * @return string
+     */
+    protected function getHostsFile()
+    {
+        if (strpos(strtoupper(PHP_OS), 'WIN') === 0) {
+            // ToDo: Someone should test this! I don't have a windows machine.
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: again

File: src/HostCommand.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        $this->hostFile = $this->getHostsFile();
+
+        if ($this->hostFile === false) {
+            $output->writeln('We could not find your host file.');
+            return null;
+        }
+
+
+        if ($this->dialog->askConfirmation($output,
+            "Mapping '" . $this->ip . "' to '" . $this->hostName . "'. Is that okay? (Y/n) ")) {
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: cs

File: src/HostCommand.php [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            $output->writeln('Okay! Will now write to hosts file...');
+
+            $command = $this->getCommand();
+
+            if ($command === false) {
+                $output->writeln('We could not find the correct command.');
+                return null;
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: just `return;`, also, you need a newline before this statement

File: src/HostCommand.php [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    /**
+     * Execute the command.
+     *
+     * @param  \Symfony\Component\Console\Input\InputInterface  $input
+     * @param  \Symfony\Component\Console\Output\OutputInterface  $output
+     * @return void
+     */
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var DialogHelper $dialog */
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: no

File: src/HostCommand.php [hunk 7]
```diff
@@ -0,0 +1,131 @@
+<?php
+namespace Laravel\Homestead;
+
+use Symfony\Component\Console\Helper\DialogHelper;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Process\Process;
```
Comment (GrahamCampbell) [inline src/HostCommand.php]: order these by length

File: composer.lock [hunk 8]
```diff
@@ -1,32 +1,32 @@
 {
     "_readme": [
         "This file locks the dependencies of your project to a known state",
-        "Read more about it at http://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
+        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
```
Comment (GrahamCampbell) [inline composer.lock]: Please revert all changes to the lock file.

**Comment by Cyruxx** [comment]
Hey Graham, what do you mean with "cs"?

I'll make another commit if you're fine with this idea.

**Comment by GrahamCampbell** [comment]
Code style.

> I'll make another commit if you're fine with this idea.

Not up to me.

**Comment by GrahamCampbell** [comment]
Only Taylor can decide if he likes this.

**Comment by Cyruxx** [comment]
Allright. Thanks! :-)

File: src/HostCommand.php (hunk 1 above)
Comment (Cyruxx) [inline src/HostCommand.php]: The statements look good, I've copied this from the EditCommand.

**Comment by Cyruxx** [comment]
Updated.

**Comment by svpernova09** [comment]
IMHO this shouldn't be something Homestead does. There are vagrant plugins that handle this for you automatically. My preference would be to avoid having homestead every do anything that would require sudo.

**Pull request closed without merging**